src/main/java/org/dataone/hashstore/HashAddress.java

```diff
@@ -0,0 +1,84 @@
+package org.dataone.hashstore;
+
+import java.nio.file.Path;
+import java.util.Map;
+
+/**
+ * HashAddress is a class that models a unique identifier for a file in the
+ * Hashstore. It encapsulates information about the file's name, path, and
+ * associated hash digest values. By using HashAddress objects, client code can
+ * easily locate, retrieve, and modify files in the HashStore without needing to
+ * know the underlying file system details.
```

Comment:
Thank you for sharing Hyrum's article - that makes complete sense. The absolute and relative path was a pattern carried over during the prototyping phase - and I left it in because I thought it might be helpful to have. However, it's apparent now that this is unnecessary (and potentially dangerous). I will remove these two implementation details and refactor accordingly.